src/js/dqa.jsx

```diff
@@ -21,122 +15,212 @@ var SortTypes = {
+
+                    rows.push({
+                        code: code,
+                        desc: desc,
+                        status: status,
+                        sites: sites,
+                    });
+                }
             }
         }
         return rows;
     },
 
+    componentWillReceiveProps: function(nextProps) {
+        this.setState({
+            rows: this._getRows(nextProps),
+        })
+    },
+
+    componentWillMount: function() {
+        this.setState({
+            rows: this._getRows(this.props),
+        })
+    },
+
     componentDidMount: function() {
-        this._sortRowsBy(0);
+        // set a timeout for updating width, otherwise everything on the page gets
+        // re-mounted on any navigation. I don't quite understand why, but somehow 
+        // this is caused by the fact that _updateWidth accesses dimentions of an element.
+        setTimeout(this._updateWidth);
+
+        if (window.addEventListener) {
+          window.addEventListener('resize', this._onResize, false);
+        } else if (window.attachEvent) {
+          window.attachEvent('onresize', this._onResize);
+        } else {
+          window.onresize = this._onResize;
+        }
+    },
+
+    _onResize: function() {
+        this._updateWidth();
+    },
+
+    _updateWidth: function() {
+        // there has to be a better way to size the table correctly, but this will do for now.
+        // (TODO figure out a cleaner way to do this)
+        var component = document.getElementById('dqaContainer');
+        if (component) {
+            this.setState({
+                tableWidth : component.clientWidth
+            });
+        }
     },
 
     _onColumnResizeEndCallback: function(newColumnWidth, dataKey) {
-        columnWidths[dataKey] = newColumnWidth;
+        w = this.state.columnWidths
+        w[dataKey] = newColumnWidth;
+        this.setState({
+            columnWidths: w
+        })
         isColumnResizing = false;
-        this.forceUpdate(); // don't do this, use a store and put into this.state! TODO
     },
 
-    _sortRowsBy: function(cellDataKey) {
+    _sortRowsBy: function(sortBy) {
         var sortDir = this.state.sortDir;
-        var sortBy = cellDataKey;
 
+        // if we are sorting by the same column as we sorted last time,
+        // reverse the sort order;
+        // otherwise pick the default (ascending) order.
         if (sortBy === this.state.sortBy) {
           sortDir = this.state.sortDir === SortTypes.ASC ? SortTypes.DESC : SortTypes.ASC;
         } else {
-          sortDir = SortTypes.DESC;
+          sortDir = SortTypes.ASC;
         }
         
-        var rows = this.state.rows.slice();
+        var rows = this._getRows(this.props, sortBy).slice();
         rows.sort(function(a, b) {
-        var sortVal = 0;
+            var sortVal = 0;
 
-        /*
-        if (sortBy === 0) {
-            var validValues = ['high', 'medium', 'low'];
+            /*
```

Comment:
I kept this just in case we decide to put the issue rank back into the DQA report. But then it's not exactly hard to replicate if needed. I'll remove it, thanks.